Source/WebCore/html/ImageDataArray.h

```diff
@@ -28,34 +28,48 @@
 
 #pragma once
 
+#include "ImageDataStorageFormat.h"
+#include <JavaScriptCore/Float16Array.h>
 #include <JavaScriptCore/Uint8ClampedArray.h>
+#include <optional>
+#include <variant>
+#include <wtf/JSONValues.h>
 
 namespace WebCore {
 
 class ImageDataArray {
 public:
-    // FIXME: This API is the minimum subset as currently used by ImageData. More to come...
-    ImageDataArray(Ref<JSC::Uint8ClampedArray>&& data)
-        : m_bufferView(WTFMove(data))
-    { }
-
-    auto length() const
-    {
-        Ref bufferView = m_bufferView;
-        return bufferView->length();
-    }
-
-    // FIXME: Remove these temporary functions.
-    Uint8ClampedArray& get() const { return m_bufferView.get(); }
-    auto data() const
-    {
-        Ref bufferView = m_bufferView;
-        return bufferView->data();
-    }
+    static constexpr bool isSupported(JSC::TypedArrayType);
+    static constexpr bool isSupported(const JSC::ArrayBufferView&);
+
+    ImageDataArray(Ref<JSC::Uint8ClampedArray>&&);
+    ImageDataArray(Ref<JSC::Float16Array>&&);
+
+    static std::optional<ImageDataArray> tryCreateZeroFilled(size_t, ImageDataStorageFormat);
+    static std::optional<ImageDataArray> tryCreateFromBufferOrZero(size_t, ImageDataStorageFormat, const uint8_t*, size_t);
+
+    ImageDataStorageFormat storageFormat() const;
+    size_t length() const;
+
+    JSC::ArrayBufferView& arrayBufferView() const { return m_arrayBufferView.get(); }
+    JSC::ArrayBufferView* operator->() const { return &arrayBufferView(); }
```

Comment:
This operator is weird. It makes ImageDataArray act like a pointer. Usually when this operator is implemented, the `operator*` and the `operator bool` are also implemented. So if your class allows you to say `foo->bar`, it should allow you to say `*foo` and `if (foo)`. But this does not happen here. `std::optional` is an example.

Reply:
I'll remove it, it shouldn't be needed anymore after I've added pass-through functions like `span()` as you suggested/requested.
And if there's still a need to access the `ArrayBufferView`, `arrayBufferView()` is there for that.